src/LaunchDarkly.EventSource/EventSourceService.cs

```diff
@@ -228,46 +228,23 @@ private HttpRequestMessage CreateHttpRequestMessage(Uri uri, string lastEventId)
             return request;
         }
 
-        private void HandleInvalidResponses(HttpResponseMessage response)
+        private void ValidateResponse(HttpResponseMessage response)
         {
-            HandleUnsuccessfulStatusCodes(response);
-
-            // According to Specs, a client can be told to stop reconnecting using the HTTP 204 No Content response code
-            HandleNoContent(response);
-
-            // According to Specs, HTTP 200 OK responses that have a Content-Type specifying an unsupported type, 
-            // or that have no Content-Type at all, must cause the user agent to fail the connection.
-            HandleIncorrectMediaType(response);
```

Comment:
I cleaned up this logic a bit just to eliminate some extra helper functions that weren't really necessary, and to make it easier to see all the rules in one place.